enhancements/cert-manager/istio-csr-controller.md

```diff
@@ -0,0 +1,878 @@
+---
+title: istio-csr-controller
+authors:
+  - "@bhb"
+reviewers:
+  - "@tgeer" ## reviewer for cert-manager component
+approvers:
+  - "@tgeer" ## approver for cert-manager component
+api-approvers:
+  - "@tgeer" ## approver for cert-manager component
+creation-date: 2024-01-22
+last-updated: 2024-05-13
+tracking-link:
+  - https://issues.redhat.com/browse/CM-234
+see-also:
+  - NA
+replaces:
+  - NA
+superseded-by:
+  - NA
+---
+
+# Extend cert-manager-operator to manage istio-csr.
+
+## Summary
+
+This enhancement describes the proposal to extend cert-manager-operator to deploy and manage the `istio-csr`
+solution. `OpenShift Service Mesh` facilitates the security of both intra-cluster and external service
+communications through mTLS. And istio-csr acts as an agent to obtain and renew certificates that
+facilitate mTLS from the `cert-manager`.
+
+## Motivation
+
+cert-manager enables obtaining certificates from [various](https://docs.openshift.com/container-platform/4.15/security/cert_manager_operator/index.html#cert-manager-issuer-types_cert-manager-operator-about)
+certificate authorities and also ensures the validity of certificates by renewing them before they expire.
+`OpenShift Service Mesh` secures intra-cluster and external service connections with TLS encryption. `cert-manager`
+can be used to obtain the required certificates through custom resources `issuer` and `clusterissuer`. `istio-csr`
+is an agent that simplifies the process of obtaining the certificates from the `cert-manager` for
+`OpenShift Service Mesh`. `cert-manager-operator` manages `cert-manager` and extending the operator to manage
+`istio-csr` will help the users to use all the solutions mentioned above effectively and easily.
+
+### User Stories
+
+- As an OpenShift user, I want to have an option to dynamically enable `istio-csr`, so that it can be used only
```

Comment:
A user story is required for installation, un-installation and upgrade usecases.

Reply:
comment incorporated.